src/app/clusters/door-lock-server/door-lock-server.h

```diff
@@ -252,6 +259,63 @@ class DoorLockServer
 
     DlLockDataType credentialTypeToLockDataType(DlCredentialType credentialType);
 
+    /**
+     * @brief Common handler for LockDoor, UnlockDoor, UnlockWithTimeout commands
+     *
+     * @param commandObj    original command context
+     * @param commandPath   original command path
+     * @param opType        remote operation type (lock, unlock)
+     * @param opHandler     plugin handler for specified command
+     * @param pinCode       pin code passed by client
+     * @return true         if locking/unlocking was successfull
+     * @return false        if error happenned during lock/unlock
+     */
+    bool HandleRemoteLockOperation(chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+                                   DlLockOperationType opType, RemoteLockOpHandler opHandler,
+                                   const chip::Optional<chip::ByteSpan> & pinCode);
+
+    /**
+     * @brief Send LockOperation event if opSuccess is true, otherwise send LockOperationError with given opErr code
+     *
+     * @param endpointId    endpoint where DoorLockServer is running
+     * @param opType        lock operation type (lock, unlock, etc)
+     * @param opSource      operation source (remote, keypad, auto, etc)
+     * @param opErr         operation error code (if opSuccess == false)
+     * @param userId        user id
+     * @param fabricIdx     fabric index
+     * @param nodeId        node id
+     * @param credList      list of credentials used in lock operation (can be NULL if no credentials were used)
+     * @param credListSize  size of credentials list (if 0, then no credentials were used)
+     * @param opSuccess     flags if operation was successfull or not
+     */
+    void SendLockOperationEvent(chip::EndpointId endpointId, DlLockOperationType opType, DlOperationSource opSource,
+                                DlOperationError opErr, Nullable<uint16_t> userId, Nullable<chip::FabricIndex> fabricIdx,
+                                Nullable<chip::NodeId> nodeId, LockOpCredentials * credList, size_t credListSize,
+                                bool opSuccess = true);
+
+    /**
+     * @brief Send generic event
+     *
+     * @tparam T            Any event type supported by Matter
+     * @param endpointId    endpoint where DoorLockServer is running
+     * @param event         event object built by caller
+     */
+    template <typename T>
+    void SendEvent(chip::EndpointId endpointId, T & event);
+
+    friend bool
+    emberAfDoorLockClusterLockDoorCallback(chip::app::CommandHandler * commandObj,
```

Comment:
Is there a reason to have function below as a `friend`? As far as I can tell their 'friendliness' is not used anywhere and I don't think I can come up with a way to use it. Could you prvide an example when this could be helpful?

Reply:
`LockDoor`, `UnlockDoor` and `UnlockWithTimeout` commands share the same handler which is private in `DoorLockServer` class. Also it calls method that send cluster events. So _friendiness_ is necessary by 2 reasons:

1. We decided to incapsulate all cluster stuff in a class;
2. Mentioned private method should be private as it is handles only remote commands from client. It is useless for the server-side app.
If 1 is not actual anymory we can avoid _friendiness_ and implement private methods as module-local functions.